File: Selenium/Project/Activity7.py
from threading import Thread
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver import ActionChains
from selenium.webdriver.support import expected_conditions as ExpectedConditions
from selenium.webdriver.support.wait import WebDriverWait


#create WebDriver instance
with webdriver.Firefox() as driver:

    # Declare the wait variable
    wait = WebDriverWait(driver, timeout=10)

    builder = ActionChains(driver)

    #go to web page
    driver.get("http://alchemy.hguy.co/orangehrm")

    #do the stuff on the page
    pageTitle = driver.title
    print("Title of the page is : " + pageTitle)

#--------------------------------------------------------------------------------------------------
    
    # enter credentials in the username and password fields
    driver.find_element(By.XPATH, "//*[@id=\"txtUsername\"]").send_keys("orange")
	
    driver.find_element(By.XPATH, "//*[@id=\"txtPassword\"]").send_keys("orangepassword123")
	
	#click submit button
    driver.find_element(By.XPATH, "//*[@id=\"btnLogin\"]").click()
    
    # Print the title of the new page
    pageTitle = driver.title
    print("New page title is: ", pageTitle)

    # Read login message
    print("Confirmation message is - " , driver.find_element(By.ID, "welcome").text)

	#Print the confirmation message
    message = driver.find_element(By.TAG_NAME, "h1")
    print("You are on : ", message.text)
    
#--------------------------------------------------------------------------------------------------
    #Add employee qualifications

    #Find the “My Info” menu item and click it.
    #driver.find_element(By.XPATH, "//a[@id='menu_pim_viewMyDetails']").click()

    leave = driver.find_element(By.XPATH, "//*[@id=\"menu_pim_viewMyDetails\"]")

    builder.move_to_element(leave).pause(5).click(leave).pause(10).perform()
    print("Clicked on My Info tab")

    message = driver.find_element(By.XPATH, "/html/body/div[1]/div[3]/div/div[2]/div[1]/h1").text
    print("You are on " + message)

    #On the new page, find the Qualification option on the left side menu and click it.
    driver.find_element(By.XPATH, "/html/body/div[1]/div[3]/div/div[1]/ul/li[9]/a").click()
    print("Clicked on Qualifications")

    #Add Work Experience and click Save.
    driver.find_element(By.XPATH, "//input[@id='addWorkExperience']").click()
    print("Clicked on addWorkExperience")

    message = driver.find_element(By.CSS_SELECTOR, "div[id='changeWorkExperience'] div[class='head']").text
    print("You are on " + message)

    driver.find_element(By.ID, "experience_employer").clear()
    driver.find_element(By.ID, "experience_employer").send_keys("AAA Test Company")

    driver.find_element(By.ID, "experience_jobtitle").clear()
    driver.find_element(By.ID, "experience_jobtitle").send_keys("Automation Tester")

    driver.find_element(By.ID, "experience_from_date").clear()
    driver.find_element(By.ID, "experience_from_date").send_keys("2021-07-01")

    driver.find_element(By.ID, "experience_comments").clear()
    driver.find_element(By.ID, "experience_comments").send_keys("Total 3 years of experience.")

    driver.find_element(By.XPATH, "//input[@id='btnWorkExpSave']").click()
    print("Work Experience Information saved successfully..!!")

#--------------------------------------------------------------------------------------------------

    #close the browser
    driver.quit()

File: Selenium/Project/Activity7.java
package seleniumproject;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

public class Activity7 
{
	WebDriver driver;
	String message;
	WebDriverWait wait;
	Actions builder;
	
		
	@BeforeClass
	public void setup() 
    {
		//create a WebDriver instance
		driver = new FirefoxDriver();
		
		builder = new Actions(driver);
		
		//go to a web page
		driver.get("http://alchemy.hguy.co/orangehrm");

		//do stuff on the page
		System.out.println("Title of the page is : " + driver.getTitle());
    }
	
	@BeforeTest
	public void setupBeforeTest() 
    {
		// Initialize wait
        wait = new WebDriverWait(driver, Duration.ofSeconds(60));		
    }
 
	@AfterClass
	public void tearDown() 
	{
 	     // Close the driver
		driver.close();
	}
	
	@DataProvider(name = "LoginCredientials")
	public static Object[][] credentials() {
	    return new Object[][] { 
	        { "orange", "orangepassword123" }
	    };
	}
	
	@Test(priority = 1, dataProvider = "LoginCredientials")
    public void login(String sUsername, String sPassword) 
    {
		// Find the username and password fields
        WebElement username = driver.findElement(By.id("txtUsername"));
        WebElement password = driver.findElement(By.name("txtPassword"));

        // Enter credentials
        username.sendKeys(sUsername);
        password.sendKeys(sPassword);

        // Click login
        driver.findElement(By.xpath("//*[@id=\"btnLogin\"]")).click();

        System.out.println("Title of the new page is : " + driver.getTitle());
        
        // Read login message
        String loginMessage = driver.findElement(By.id("welcome")).getText();
        
        System.out.println("Confirmation message is - " + loginMessage);
                
        //Verify that the homepage has opened
        message = driver.findElement(By.tagName("h1")).getText();
        System.out.println("You are on " + message);
        
        Assert.assertEquals(message, "Dashboard");
    }
	

	@Test(priority = 2)
    public void addQualifications() 
    {
		//Add employee qualifications

		//Find the “My Info” menu item and click it.
		//driver.findElement(By.xpath("//a[@id='menu_pim_viewMyDetails']")).click();
			
		WebElement leave = driver.findElement(By.xpath("//*[@id=\"menu_pim_viewMyDetails\"]"));
		
		builder.moveToElement(leave).pause(1000).click(leave).pause(2000).build().perform();
		System.out.println("Clicked on My Info tab");
		
		message = driver.findElement(By.xpath("/html/body/div[1]/div[3]/div/div[2]/div[1]/h1")).getText();
		System.out.println("You are on " + message);
        
		//On the new page, find the Qualification option on the left side menu and click it.
		driver.findElement(By.xpath("/html/body/div[1]/div[3]/div/div[1]/ul/li[9]/a")).click();
		System.out.println("Clicked on Qualifications");
		
		//Add Work Experience and click Save.
		driver.findElement(By.xpath("//input[@id='addWorkExperience']")).click();
		System.out.println("Clicked on addWorkExperience");
		
		message = driver.findElement(By.cssSelector("div[id='changeWorkExperience'] div[class='head']")).getText();
		System.out.println("You are on " + message);
		
		 driver.findElement(By.id("experience_employer")).sendKeys("AAA Test Company");
		 driver.findElement(By.id("experience_jobtitle")).sendKeys("Automation Tester");
		 driver.findElement(By.id("experience_from_date")).sendKeys("2021-07-01");
		 driver.findElement(By.id("experience_comments")).sendKeys("Total 3 years of experience.");
	     
		 driver.findElement(By.xpath("//input[@id='btnWorkExpSave']")).click();
		 System.out.println("Work Experience Information saved successfully..!!");
    }

}
